File: gfapy/line/segment/writer_wo_sequence.py
class WriterWoSequence:

  def __str__(self, without_sequence = False):
    """
    Parameters
    ----------
    without_sequence : bool
    	 If **True**, output "*" instead of sequence.
    
    Returns
    -------
    str
    	String representation of the segment.
    """
    if not without_sequence:
      return super().__str__()
    else:
      saved = self.sequence
      self.sequence = "*"
      retval = super().__str__()
      self.sequence = saved
      return retval

File: lib/rgfa/line/segment/writer_wo_sequence.rb
require_relative "../segment"

module RGFA::Line::Segment::WriterWoSequence

  # @return string representation of the segment
  # @param [Boolean] without_sequence if +true+, output "*" instead of sequence
  def to_s(without_sequence: false)
    if !without_sequence
      return super()
    else
      saved = self.sequence
      self.sequence = "*"
      retval = super()
      self.sequence = saved
      return retval
    end
  end

end
